server/graphql/inventory_adjustment/mutations/insert.rs

```diff
@@ -0,0 +1,44 @@
+use async_graphql::*;
+use graphql_core::{standard_graphql_error::validate_auth, ContextExt};
+use graphql_types::types::InvoiceNode;
+use repository::Invoice;
+use service::auth::{Resource, ResourceAccessRequest};
+
+#[derive(InputObject)]
+#[graphql(name = "CreateInventoryAdjustmentInput")]
+pub struct CreateInventoryAdjustmentInput {
+    pub stock_line_id: String,
+    pub new_number_of_packs: f64,
+    pub inventory_adjustment_reason_id: Option<String>,
+}
+
+#[derive(Union)]
+#[graphql(name = "CreateInventoryAdjustmentResponse")]
+pub enum InsertResponse {
+    Response(InvoiceNode),
+}
+
+pub fn create_inventory_adjustment(
+    ctx: &Context<'_>,
+    store_id: &str,
+    _input: CreateInventoryAdjustmentInput,
+) -> Result<InsertResponse> {
+    let user = validate_auth(
+        ctx,
+        &ResourceAccessRequest {
+            // TODO map Permissions::EnterInventoryAdjustments
+            resource: Resource::CreateRepack,
+            store_id: Some(store_id.to_string()),
+        },
+    )?;
+
+    let service_provider = ctx.service_provider();
+    let _service_context = service_provider.context(store_id.to_string(), user.user_id)?;
+
+    // another pattern to discuss - returning data from mutation when we don't use it?
```

Comment:
Maybe we can import some bare minimum thing ?

I would suspect the main thing that needs to be returned is 'new number of packs', but we won't really use it, and we do need some kind of return, not sure, maybe we have a specific type we can easily return, similar to some of the common structure errors